drgn_tools/kvm.py

```diff
@@ -183,6 +182,70 @@ def print_vcpu_list(prog: Program) -> None:
     print_table(rows)
 
 
+def print_memslot_info(prog: Program) -> None:
+    vm_list = for_each_vm(prog)
+    rows = [
+        [
+            "KVM",
+            "KVM_MEMSLOTS",
+            "KVM_MEMORY_SLOT",
+            "BASE_GFN",
+            "PAGES",
+            "ARCH",
+            "USER_ADDR",
+            "FLAGS",
+        ]
+    ]
+
+    for vm in vm_list:
+        for i in range(len(vm.memslots)):
+            kvm_memslots = hex(vm.memslots[i])
+            if has_member(vm.memslots[i], "memslots"):
+                for j in range(vm.memslots[i].used_slots.value_()):
```

Comment:
Similarly, it would be slightly better to do:
```suggestion
                for j in range(vm.memslots[i].used_slots.value_()):
                    mm = memslots[j]
```
And then use `mm` (or a better variable name) in place of `vm.memslots[i].memslots[j]` below.

Reply:
Thank you for the suggestion Stephen. I have made the changes accordingly.